File: src/main/scala/scala_code/Words.scala
package scala_code

import scala.io.Source

object Words {
  private val WORDS_FILE = "words"

  def main(args: Array[String]): Unit = {
    val source = Source.fromFile(WORDS_FILE)
    source.getLines()
      .filter(_.length() > 16)
      .toList
      .sorted((word1: String, word2: String) => word1.length - word2.length)
      .map(word => word.length + ": " + word)
      .foreach(println(_))
  }
}

File: src/main/java/java_code/Words.java
package java_code;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Words {

    private static final Path WORDS_FILE = Path.of("words");

    public static void main(String[] args) throws IOException {
        Files.lines(WORDS_FILE)
                .filter(word -> word.length() > 16)
                .sorted((w1, w2) -> w1.length() - w2.length())
                .map(word -> word.length() + ": " + word)
                .forEach(word -> System.out.println(word));
    }
}
